client/src/redux/info/infoReducer.js

```diff
@@ -30,14 +32,14 @@ export default function infoReducer(state = defaultState, action) {
 
       return newState;
     }
-    case actionTypes.UPDATE_STARTING_ROWS: {
+    case actionTypes.UPDATE_AVAILABLE_ROWS: {
       const newState = { ...state };
-      newState.startingRows.push(action.row);
+      newState.availableRows = action.availableRows;
       return newState;
     }
-    case actionTypes.RESET_STARTING_ROWS: {
+    case actionTypes.RESET_AVAILABLE_ROWS: {
       const newState = { ...state };
-      newState.startingRows = [];
+      newState.availableRows = shuffle(Array(constants.GRID_SIZE).fill(null).map((_, i) => (i)));
```

Comment:
extract making a new shuffled array into a helper function so that it can be used here and to define the initial state